### DISCUSSION TITLE
v0.2.0API修正

### DISCUSSION THREAD

**Discussion by namikis** [jphacks/F_2109#56] [General]
@kenkenpa2126 @nishidayoshikatsu 
仕様の関係で、以下の三つのAPI設計を変更したいです。
 
---------------------------------------------------------------------------

POST /v0.2.0/upload
request
{
  "user_id": number
  "problem_id": number
  "source_code" : Dir
  "browser": string （今はChromeのみ）
}

↓

{
  "user_id": number
  "problem_image": string
  "source_code" : Dir
  "browser": string （今はChromeのみ）
}

---------------------------------------------------------------------------

POST /v0.2.0/imgScore
{
"url" : string
}

↓

request
{
"user_id": number,
"problem_image": string,
"url" : string
}

---------------------------------------------------------------------------

POST /v0.2.0/close
request
{
"id" : string        // testuser_p1
}

↓

{
"user_id": number,
"problem_image": string
}

**Comment by nisshi-dev** [comment]
@namikis 
POST /v0.2.0/upload
request
{
"user_id": number
"problem_image": string
"source_code" : Dir
"browser": string （今はChromeのみ）
}

命名的な話ですが、problem_imageだと、問題画像そのものだと思ってしまうので、problem_image_nameとかの方が混乱しにくいですかね

**Reply by kenkenpa2126** [reply]
そういうことか。てっきり画像そのもの送ってるんかと思った。

**Reply by namikis** [reply]
たしかにそうですね。すみません。
@kenkenpa2126 命名の部分も含めてよろしくお願いいたします。

**Comment by kenkenpa2126** [comment]
@namikis 
これ、ProblemImageNameじゃなくて、problemIdでいいですか？

**Reply by kenkenpa2126** [reply]
変更前のproblem_idとproblem_imageの違いはなんですか？

**Reply by namikis** [reply]
@kenkenpa2126 
problem_idはDBに挿入されたときに自動で追加される数値です。
この値はデータを入れ替えたりするたびに変動するので、画像名は別で付けといたほうがいいということでproblem_imageを使っています。

**Reply by kenkenpa2126** [reply]
problem_idをinteger primary keyにしたってことですね。
p1とかp2とかもidに近いので、そっちをproblemIdにして、integer primary keyはidって名前にするか、もしくは設定しなくていいと思います。（面倒ならこのままproblemImageNameでもいいともいます。）
integer primary keyはどういうときに使ってますか？

**Reply by namikis** [reply]
@kenkenpa2126 
integer primary keyは/v0.2.0/problem/detailのリクエストとかがそうですね。一見不要に見えますが分けておいた方が管理がしやすいので、このままproblemImageNameでお願いします！

**Reply by namikis** [reply]
@kenkenpa2126 
ちなみにproblem_idってproblemテーブルのidカラムっていう認識であってますよね？
